.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateTimePeriodExtractor.cs

```diff
@@ -637,5 +672,41 @@ private List<Token> MatchRelativeUnit(string text)
 
             return ret;
         }
+
+        private Token MatchWithinNextPrefix(string subStr, string text, Token duration, bool inPrefix)
+        {
+            var startOut = -1;
+            var endOut = -1;
+            bool success = false;
+            var match = config.WithinNextPrefixRegex.Match(subStr);
+            if (MatchPrefixRegexInSegment(subStr, match, inPrefix))
+            {
+                var startToken = inPrefix ? match.Index : duration.Start;
+                var endToken = duration.End + (inPrefix ? 0 : match.Index + match.Length);
+                match = config.TimeUnitRegex.Match(text.Substring(duration.Start, duration.Length));
+                success = match.Success;
+
+                if (!inPrefix)
+                {
+                    // Match prefix for "next"
+                    var beforeStr = text.Substring(0, duration.Start);
+                    var matchNext = this.config.NextPrefixRegex.Match(beforeStr);
+                    success = match.Success || matchNext.Success;
+                    if (MatchPrefixRegexInSegment(beforeStr, matchNext, true))
+                    {
+                        startToken = matchNext.Index;
+                    }
+                }
+
+                if (success)

```

Comment:
What happens if not a success?

Reply:
if not a success, the method returns a token that starts and ends at -1. When the method is called, it is checked if the returned token.Start is >=0.